optimum/bettertransformer/models/attention.py

```diff
@@ -89,6 +89,35 @@ def gpt2_wrapped_scaled_dot_product(
     return sdpa_result, None
 
 
+# Adapted from transformers.models.bark.modeling_bark.BarkSelfAttention._attn
+def bark_wrapped_scaled_dot_product(
+    self,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    attention_mask: Optional[torch.Tensor] = None,
+    head_mask: Optional[torch.Tensor] = None,
+):
+    raise_on_head_mask(head_mask)
+
+    # When `past_kv` is provided, we're doing incremental decoding and `q.shape[2] == 1`: q only contains
+    # the query for the last token. scaled_dot_product_attention interprets this as the first token in the
+    # sequence, so if is_causal=True it will mask out all attention from it. This is not what we want, so
+    # to work around this we set is_causal=False.
+    is_causal = self.is_causal and query.shape[2] != 1
+
+    if is_causal:
+        sdpa_result = torch.nn.functional.scaled_dot_product_attention(
+            query, key, value, attn_mask=None, dropout_p=self.dropout, is_causal=is_causal
+        )
+    else:
+        sdpa_result = torch.nn.functional.scaled_dot_product_attention(
+            query, key, value, attn_mask=attention_mask, dropout_p=self.dropout, is_causal=False
```

Comment:
The point of the batch size / training & eval check in the other functions is that SDPA does not dispatch to mem efficient attention / flash attention in case a custom attention mask is passed.

So here with this current implem, if somebody fine-tunes bark (which typically does not involve any padding, thus no need of custom attention mask) with batch size > 1, there would be close to no speedup.

Is there anything special that makes passing an attention mask required?

Reply:
I didn't know about the use of the training & eval check. What about the point of the `batch_size == 1` check?

At the moment, Bark is only used for inference, but I could probably change the code with regards to the training & eval check, just in case training is supported in the future.

The attention mask is not really used in `BarkModel` yet but the underlying sub-models might require it for finetuning in the future I guess.